src/test/java/tests/logger/LoggerTests.java

```diff
@@ -0,0 +1,177 @@
+package tests.logger;
+
+import org.apache.log4j.*;
+import org.testng.annotations.*;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.UUID;
+import aquality.selenium.core.logging.Logger;
+import tests.application.CustomAqualityServices;
+
+import static org.testng.Assert.*;
+
+public class LoggerTests {
+    private final static String appenderLogFilePattern = "target/log/appender-%s.log";
+    private final static String testMessage = "test message";
+    private final static String testExceptionText = "test exception";
+    private final static String log4jFieldName = "log4J";
+    private Logger logger;
+    private org.apache.log4j.Logger log4j;
+    private Appender appender;
+    private File appenderFile;
+
+    @BeforeMethod
+    private void addMessagesAppender() throws IOException {
+        logger = CustomAqualityServices.getInjector().getInstance(Logger.class);
+        appenderFile = getRandomAppenderFile();
+        appender = getFileAppender(appenderFile);
+        logger.addAppender(appender);
+    }
+
+    @BeforeGroups("messages")
+    private void initializeLog4jField() throws NoSuchFieldException, IllegalAccessException {
+        logger = CustomAqualityServices.getInjector().getInstance(Logger.class);
+        Field log4jField = Logger.class.getDeclaredField(log4jFieldName);
+        log4jField.setAccessible(true);
+        log4j = ((ThreadLocal<org.apache.log4j.Logger>) log4jField.get(logger)).get();
+    }
+
+    @Test
+    public void testAqualityServicesShouldReturnInstanceofLogger() {
+        assertEquals(logger, Logger.getInstance());
+    }
+
+    @Test
+    public void testShouldBePossibleToAddAppender() throws IOException {
+        logger.addAppender(appender).info(testMessage);
+        assertTrue(appenderFile.exists(), String.format("New appender is not added to log4j. File '%s' is not created.", appenderFile.getPath()));
```

Comment:
rephrase message with should
here and in all tests below

Reply:
updated